engine/src/main/java/org/terasology/engine/world/chunks/localChunkProvider/LocalChunkProvider.java

```diff
@@ -437,10 +405,101 @@ private boolean isChunkReady(Chunk chunk) {
         return chunk != null && chunk.isReady();
     }
 
+    public void notifyRelevanceChanged() {
+        loadingPipeline.notifyUpdate();
+    }
+
+    private void updateList() {
+        relevanceSystem.neededChunks()
+                .filter(pos -> !chunksInRange.contains(pos))
+                .forEach(chunksInRange::add);
+        chunksInRange.removeIf(x -> !relevanceSystem.isChunkInRegions(x) || isChunkReady(x));
+        chunksInRange.sort(relevanceSystem.createChunkPosComparator().reversed());
```

Comment:
I don't think so, the only place we add to `chunksInRange` is the line above which `filter`s out duplicates.

Reply:
Oh wait, duplicates actually are possible when there are multiple clients connected with overlapping relevance regions, since there will be duplicates in `neededChunks`. However, that shouldn't actually cause any problems because the code to generate chunks checks `currentlyProcessing` first.